src/Symfony/Component/ExpressionLanguage/Node/CoalescingNode.php

```diff
@@ -0,0 +1,60 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\ExpressionLanguage\Node;
+
+use Symfony\Component\ExpressionLanguage\Compiler;
+
+/**
+ * @author Fabien Potencier <fabien@symfony.com>
+ *
+ * @internal
+ */
+class CoalescingNode extends Node
+{
+    public function __construct(Node $expr1, Node $expr2)
+    {
+        parent::__construct(
+            ['expr1' => $expr1, 'expr2' => $expr2]
+        );
+    }
+
+    public function compile(Compiler $compiler)
+    {
+        $compiler
+            ->raw('((')
+            ->compile($this->nodes['expr1'])
+            ->raw(') ?? (')
+            ->compile($this->nodes['expr2'])
+            ->raw('))')
+        ;
+    }
+
+    public function evaluate(array $functions, array $values)
+    {
+        try {
+            $expr1 = @$this->nodes['expr1']->evaluate($functions, $values);
```

Comment:
we should find a way to not have to use the `@` operator, especially because future versions of PHP are turning many such warnings into real `Error`

Reply:
This is the most challenging part!